test/EndToEnd/tests/PackageNameSpaceTests.ps1

```diff
@@ -326,6 +326,296 @@ function Test-VsPackageInstallerServices-PackageNamespaceInstall-WithMultipleFee
+        Assert-AreEqual 0 $errorlist.Count
+        $warninglist = Get-Warnings
+        Assert-AreEqual 0 $warninglist.Count
+    }
+    finally {
+        Remove-Item $nugetConfigPath
+    }
+}
+
+function Test-PackageNamespaceInstall_WrongSource_Fails
+{
+    param($context)
+
+    # Arrange
+    $repoDirectory = $context.RepositoryRoot
+    $privateRepo = Join-Path $repoDirectory "privateRepo"
+
+    $nugetConfigPath = Join-Path $OutputPath 'nuget.config'    
+	$settingFileContent =@"
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <packageSources>
+    <clear />
+    <add key="LocalRepository" value="{0}" />
+    <add key="PrivateRepository" value="{1}" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key="PrivateRepository">
+            <namespace id="Solution*" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>
+"@
+    try {
+        # We have to create config file before creating solution, otherwise it's not effective for new solutions.
+        $settingFileContent -f $context.RepositoryRoot,$privateRepo | Out-File -Encoding "UTF8" $nugetConfigPath
+
+        $p = New-ConsoleApplication
+
+        # Act & Assert
+        # Even though SolutionLevelPkg package exist in $repoDirectory since package namespace filter set SolutionLevelPkg can be restored only from PrivateRepository repository so it'll fail.
+        $exceptionMessage = "Package 'SolutionLevelPkg 1.0' is not found in the following primary source(s): '" + $context.RepositoryRoot + "," + $privateRepo + "'. Please verify all your online package sources are available (OR) package id, version are specified correctly."
+        Assert-Throws { $p | Install-Package SolutionLevelPkg -Version 1.0 } $exceptionMessage
+        Assert-NoPackage $p SolutionLevelPkg 1.0.0
+    }
+    finally {
+        Remove-Item $nugetConfigPath
+    }
+}
+
+function Test-PackageNamespaceInstall_NonExistingSource_Fails
+{
+    param($context)
+
+    # Arrange
+    $nugetConfigPath = Join-Path $OutputPath 'nuget.config'    
+	$settingFileContent =@"
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <packageSources>
+    <clear />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key="UnavailableRepository">
+            <namespace id="Solution*" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>
+"@
+    try {
+        # We have to create config file before creating solution, otherwise it's not effective for new solutions.
+        $settingFileContent -f $context.RepositoryRoot | Out-File -Encoding "UTF8" $nugetConfigPath
+
+        $p = New-ConsoleApplication
+
+        # Act & Assert
+        # Even though SolutionLevelPkg package exist in $repoDirectory since package namespace filter set SolutionLevelPkg can be restored only from UnavailableRepository repository so it'll fail.
+        $exceptionMessage = "Argument cannot be null or empty`r`nParameter name: primarySources"        
```

Comment:
Yes, you're right it's not really specific to 'Package namespace', it's coming from here: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L1668-L1673

Since this test is only adding confusion I'm removing it.